src/main/java/org/prebid/server/bidder/brightroll/BrightrollBidder.java

```diff
@@ -170,34 +171,66 @@ private static boolean isImpValid(Imp imp, List<BidderError> errors) {
      * Updates {@link Imp} {@link Banner} and/or {@link Video}.
      */
     private Imp updateImp(Imp imp, PublisherOverride publisherOverride) {
+        final BigDecimal bidFloor = publisherOverride != null && publisherOverride.getBidFloor() != null
+                ? publisherOverride.getBidFloor()
+                : null;
+        final List<Integer> impBattr = publisherOverride != null && publisherOverride.getImpBattr() != null
+                ? publisherOverride.getImpBattr()
+                : null;
+
         final Banner banner = imp.getBanner();
         if (banner != null) {
-            final Banner.BannerBuilder bannerBuilder = banner.toBuilder();
-            if (banner.getW() == null && banner.getH() == null && CollectionUtils.isNotEmpty(banner.getFormat())) {
-                // update banner with size from first format
-                final Format firstFormat = banner.getFormat().get(0);
-                bannerBuilder
-                        .w(firstFormat.getW())
-                        .h(firstFormat.getH());
-            }
-            if (publisherOverride != null) {
-                bannerBuilder.battr(publisherOverride.getImpBattr());
+            final boolean noSizes = banner.getW() == null && banner.getH() == null
+                    && CollectionUtils.isNotEmpty(banner.getFormat());
+
+            if (bidFloor != null || impBattr != null || noSizes) {
+                final Imp.ImpBuilder impBuilder = imp.toBuilder();
+
+                if (bidFloor != null) {
+                    impBuilder.bidfloor(bidFloor);
+                }
+                if (impBattr != null || noSizes) {
+                    impBuilder.banner(updateBanner(banner, impBattr, noSizes));
+                }
+
+                return impBuilder.build();
             }
-            return imp.toBuilder()
-                    .banner(bannerBuilder.build())
-                    .build();
         }
+
         final Video video = imp.getVideo();
-        if (video != null && publisherOverride != null) {
-            return imp.toBuilder()
-                    .video(video.toBuilder()
-                            .battr(publisherOverride.getImpBattr())
-                            .build())
-                    .build();
+        if (video != null && (bidFloor != null && impBattr != null)) {
```

Comment:
Just to confirm. Do you expect here both `bidFloor` and `impBattr` should not be null for `video`?
If so, below checking for null is not required.

Reply:
good catch..fixed. thanks.